frameworks/Scala/finch/src/main/scala/WebServer.scala

```diff
@@ -14,25 +13,27 @@ import io.circe._
 import io.circe.generic.auto._
 import io.circe.jawn._
 import io.circe.syntax._
-import io.finch.response.EncodeResponse
 import io.circe.{Decoder, Encoder, Json}
 
 object WebServer extends App {
+  private val dateFormat: DateFormat = new SimpleDateFormat("E, dd MMM yyyy HH:mm:ss z")
 
+  import io.finch.circe._
   val json = get("json") {
-    import io.finch.circe._
-
-    case class Message(message: String)
-    Ok(Message("Hello, World!").asJson)
+    Ok(Json.obj("message" -> Json.string("Hello, World!")))
+      .withHeader("Server" -> "finch")
+      .withHeader("Date" -> dateFormat.format(new Date()))
   }
 
-  val plaintext = get("plaintext") {
-    "Hello, World!"
+  val plaintext: Endpoint[String] = get("plaintext") {
+    Ok("Hello, World!")
+      .withHeader("Server" -> "finch")
+      .withHeader("Date" -> dateFormat.format(new Date()))
+      .withContentType(Some("text/plain"))
   }
 
-  val api: Service[Request, Response] = (json :+: plaintext).toService
-
-  Await.ready(
-    Httpx.serve(":9000", api)
+  Await.ready(Http.server
+    .withCompressionLevel(0)
```

Comment:
Sorry to bother again, but can you also disable stats receiver and tracer on a server instance? (in a same way as per #1941).

```scala
.withStatsReceiver(NullStatsReceiver)
.withTracer(NullTracer)
``